Warnings when building with GCC  10.2.1 and libpcap 1.10

The building in current Debian unstable produces the following warnings:

```
gcc -DHAVE_CONFIG_H -I. -I..   -Wdate-time -D_FORTIFY_SOURCE=2 -fcommon -g -O2 -ffile-prefix-map=/PKGS/BOOKWARM/netdiscover/netdiscover=. -fstack-protector-strong -Wformat -Werror=format-security -c -o netdiscover-fhandle.o `test -f 'fhandle.c' || echo './'`fhandle.c
main.c: In function ‘main’:
main.c:240:7: warning: ‘pcap_lookupdev’ is deprecated: use 'pcap_findalldevs' and use the first device [-Wdeprecated-declarations]
  240 |       datos.interface = pcap_lookupdev(errbuf);
      |       ^~~~~
In file included from /usr/include/pcap.h:43,
                 from ifaces.h:37,
                 from main.c:40:
/usr/include/pcap/pcap.h:394:16: note: declared here
  394 | PCAP_API char *pcap_lookupdev(char *)
      |                ^~~~~~~~~~~~~~
```

Cheers,

Eriberto

--- a/src/main.c
+++ b/src/main.c
@@ -237,13 +237,20 @@ int main(int argc, char **argv)
    /* If no iface was specified, autoselect one. exit, if no one available */
    if (datos.interface == NULL)
    {
-      datos.interface = pcap_lookupdev(errbuf);
+      pcap_if_t *devices = NULL;
 
-      if (datos.interface == NULL)
-      {
-         printf("Couldn't find default device: %s\n", errbuf);
+      if (pcap_findalldevs(&devices, errbuf) != 0) {
+         printf("Couldn't find capture devices: %s\n", errbuf);
          exit(1);
       }
+
+      if (devices == NULL || devices->name == NULL) {
+         printf("Couldn't find suitable capture device: %s\n", errbuf);
+         exit(1);
+      }
+
+      datos.interface = strdup(devices->name);
+      pcap_freealldevs(devices);
    }
 
    /* Load user config files or set defaults */

